### PR TITLE
python3Packages.snowflake-connector-python: 3.0.0 -> 3.1.0

### PR DESCRIPTION
**Pull request by risicle**
## Description of changes

https://nvd.nist.gov/vuln/detail/CVE-2023-34233

Done as two commits to make it easy to cherry-pick the 3.0.2 bump alone for 23.05.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/snowflake-connector-python/default.nix [hunk 1]
```diff
@@ -48,6 +49,7 @@ buildPythonPackage rec {
   '';
 
   nativeBuildInputs = [
+    packaging
```
Comment (tjni) [inline pkgs/development/python-modules/snowflake-connector-python/default.nix:58]: Is this required? (I wasn't able to find a direct dependency on this in a build script.)

**Comment by tjni** [comment]
Result of `nixpkgs-review pr 250015` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.snowflake-sqlalchemy</li>
    <li>python310Packages.snowflake-sqlalchemy.dist</li>
    <li>python311Packages.snowflake-sqlalchemy</li>
    <li>python311Packages.snowflake-sqlalchemy.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.dbt-snowflake</li>
    <li>python310Packages.dbt-snowflake.dist</li>
    <li>python310Packages.snowflake-connector-python</li>
    <li>python310Packages.snowflake-connector-python.dist</li>
    <li>python311Packages.dbt-snowflake</li>
    <li>python311Packages.dbt-snowflake.dist</li>
    <li>python311Packages.snowflake-connector-python</li>
    <li>python311Packages.snowflake-connector-python.dist</li>
  </ul>
</details>

**Review by tjni — APPROVED** [review]

File: pkgs/development/python-modules/snowflake-connector-python/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/development/python-modules/snowflake-connector-python/default.nix:58]: https://github.com/snowflakedb/snowflake-connector-python/blob/1380e41cd2088f0c83af0cf45010d68aac96b366/setup.cfg#L55C9-L55C9

Comment (tjni) [reply pkgs/development/python-modules/snowflake-connector-python/default.nix:58]: I think that means it's required in `propagatedBuildInputs`, and you added it there, but it shouldn't be needed (at least not directly) in `nativeBuildInputs`.

Comment (risicle) [reply pkgs/development/python-modules/snowflake-connector-python/default.nix]: Cool, let's try without.

Also realised I could use the `pythonRelaxDepsHook` for `platformdirs`

**Review by matthiasbeyer — APPROVED** [review]

**Pull request merged**